Source/Core/VideoCommon/PerformanceTracker.h

```diff
@@ -33,6 +33,31 @@ class PerformanceTracker
     return (end - begin) & (MAX_DT_QUEUE_SIZE - 1);
   }
 
+  struct TimeDataPair
+  {
+  public:
+    TimeDataPair(DT duration, DT value) : duration{duration}, value{value} {}
+    TimeDataPair(DT duration) : TimeDataPair{duration, duration} {}
+    TimeDataPair() : TimeDataPair{DT::zero()} {}
+
+    TimeDataPair& operator+=(const TimeDataPair& other)
+    {
+      duration += other.duration;
+      value += other.value;
+      return *this;
+    }
+
+    TimeDataPair& operator-=(const TimeDataPair& other)
+    {
+      duration -= other.duration;
+      value -= other.value;
+      return *this;
+    }
+
+  public:
+    DT duration, value;
```

Comment:
These are both durations, the naming therefore is surprising

Reply:
the idea is like, the duration of a sample is the span of time it covers, and the value is the amount of time represented in that sample, so the naming will be a little odd. But the idea is that value could be some sort of subset of the duration. Such as with the Max Speed counter, the value would be the amount of time spent working, while the duration is the total duration, which lets you calculate max speed.